homework.md

```diff
@@ -4,110 +4,110 @@
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
+When people want to show case your code to other people, you can put it on Github as it is the best tool you can rely on.
+
+Makes it easy to contribute to your open source projects because its free and can integrate with common platforms.
+
+Track changes in your code across versions because its clearly tracked and documented.
 
 ## Question 9
 
 What happens when you `fork` a repository?
 
-You delete it
+It allows you to freely experiment wih changes without effecting the original project.
 
-## Question 10 
+## Question 10
 
-What happens when you clone a repostory?
+What happens when you clone a repository?
 
-It send it to a friend
+cloning a repository pulls down a full copy of all the repository data that Github.com has at that point in time, including all versions of every file and folder for the project.
 
 ## Question 11
 
 What is a Pull Request?
 
-When you send a file over the internet
+Pull requests let you tell others about changes you've pushed to a branch in a repository on Github and once a pull request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits before your changes are merged into the base branch.
 
 ## Question 12
 
 What is the Git History log?
 
-It lets you see into the future
+Git History log lists the commits made in that repository in reverse chronological order that is the most recent commits show up first.
 
 ## Question 13
 
 What does it meant to "push" changes to Github?
 
-You send the changes of your local repository to Github
+You push the changes of your local repository to the remote repository on Github so that if you change your project locally and want other people to have access to the changes, you must push the changes to Github.
 
 ## Question 14
 
 What is Github?
 
-A place for us to store and share our code.
+Github is a web-based interface that uses Git. the open source version control software that lets us to store and share our code and lets multiple people make seperate changes to web pages at the same time.
 
 ## Question 15
 
 What does it mean to "Commit your changes"?
 
-Send you code to Github
+As you commit changes to your project locally, you can push those changes to GitHub so that others may access them from the remote repository.
 
 ## Question 16
 
 What program do we use to write code at CodeYourFuture?
 
-Microsoft Word
+Visual Studio code
 
 ## Question 17
 
 What do you use "Commit Messages" for?
 
-To record what changes you made to the code
+To record what changes you made to the code and give a brief explantion of what you did so that a fellow coder could understand.
 
 ## Question 18
 
 Where can I look to find help with Git?
 
-The Git Cheatsheet!
+The Git Cheatsheet and your class mates via slack.
```

Comment:
And I think maybe Git documentation on official website?

Reply:
Great Point :)